nttt/tag_trimming.py

```diff
@@ -0,0 +1,109 @@
+import re
+
+# We are using a global variable because the function passed to re.sub() will only get the match as an argument.
+# We need another way of knowing the language so that we can enable language specific beahviour.
+t_lang = None
+
+# This list holds the locales identifying languages whose trailing hypens should be treated differently.
+hypen_langs = [
```

Comment:
To be honest I'm not sure that we should handle possible mistakes like `text<0>text</0>text` -> `text <0>text</0> text` because:
- we don't have an exhaustive list of languages at hand. This value doesn't contain German and maybe some other languages that we are not aware of. And if we miss a language then this code may do more harm than good to its translations
- if I understand correctly, hyphen ('-') is only one possible special case. There might be others
- this complicates the code a lot. For instance it forces to introduce a global variable
- it wasn't originally required